### PR TITLE
[Merged by Bors] - feat: `DifferentiableOn.cpow`

### PR DESCRIPTION
**Pull request by llllvvuu**
direct application of the content above it.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by llllvvuu** [comment]
(context: this mirrors [DifferentiableOn.rpow](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Analysis/SpecialFunctions/Pow/Deriv.html#DifferentiableOn.rpow) and would've golfed https://github.com/AlexKontorovich/PrimeNumberTheoremAnd/pull/22 a bit)

**Review by sgouezel — COMMENTED** [review]
Could you also add `Differentiable.cpow` and `Differentiable.const_cpow`?

File: Mathlib/Analysis/SpecialFunctions/Pow/Deriv.lean [hunk 1]
```diff
@@ -131,6 +131,22 @@ theorem DifferentiableWithinAt.const_cpow (hf : DifferentiableWithinAt ℂ f s x
   (hf.hasFDerivWithinAt.const_cpow h0).differentiableWithinAt
 #align differentiable_within_at.const_cpow DifferentiableWithinAt.const_cpow
 
+theorem DifferentiableOn.cpow (hf : DifferentiableOn ℂ f s) (hg : DifferentiableOn ℂ g s)
+    (h0 : ∀ x ∈ s, f x ∈ slitPlane) : DifferentiableOn ℂ (fun x ↦ f x ^ g x) s :=
```
Comment (Ruben-VandeVelde) [inline Mathlib/Analysis/SpecialFunctions/Pow/Deriv.lean:135]: Is this the best spelling for `h0` or should it be something like `s ⊆ f ⁻¹' slitPlane`?

Comment (sgouezel) [reply Mathlib/Analysis/SpecialFunctions/Pow/Deriv.lean:135]: It should be `MapsTo f s slitPlane`.

**Comment by sgouezel** [comment]
bors r+
Thanks!

**Pull request closed without merging**